### PR TITLE
New api format page

### PR DESCRIPTION
**Pull request by plural**
this has a ton of stuff hard-coded, but it works locally (assuming you are running the API server locally!).

I'll come back to make the config variables and whatnot, but any early feedback would be swell!

### CODE REVIEW

File: app/Resources/views/Formats/formats.html.twig [hunk 1]
```diff
@@ -13,75 +13,118 @@
...
+      $(`#${format.id}_runner_decklists`).append(`<li><a href="${Routing.generate('decklists_list', {type:'find', 'faction':'criminal', 'packs': packs})}"><span class="icon icon-criminal influence-criminal"></span> Criminal</a></li>`);
+      $(`#${format.id}_runner_decklists`).append(`<li><a href="${Routing.generate('decklists_list', {type:'find', 'faction':'shaper', 'packs': packs})}"><span class="icon icon-shaper influence-shaper"></span> Shaper</a></li>`);
+      $(`#${format.id}_runner_decklists`).append(`<li><a href="${Routing.generate('decklists_list', {type:'find', 'faction':'adam', 'packs': packs})}"><span class="icon icon-adam influence-adam"></span> Adam</a></li>`);
+      $(`#${format.id}_runner_decklists`).append(`<li><a href="${Routing.generate('decklists_list', {type:'find', 'faction':'apex', 'packs': packs})}"><span class="icon icon-apex influence-apex"></span> Apex</a></li>`);
+      $(`#${format.id}_runner_decklists`).append(`<li><a href="${Routing.generate('decklists_list', {type:'find', 'faction':'sunny-lebeau', 'packs': packs})}"><span class="icon icon-sunny-lebeau influence-sunny-lebeau"></span> Sunny Lebeau</a></li>`);
```
Comment (distributive) [inline app/Resources/views/Formats/formats.html.twig]: Would it be possible to detect (or hardcode it) to exclude the mini-factions from formats that don't include them?

File: app/Resources/views/Formats/formats.html.twig [hunk 2]
```diff
@@ -13,75 +13,118 @@
...
+
+      $(`#${format.id}`).append(`<ul id="${format.id}_cycles"></ul>`);
+      cycles.forEach(cycle => {
+        if (legalCycleIds.has(cycle.id)) {
+          $(`#${format.id}_cycles`).append(`<li><a href="${Routing.generate('cards_cycle', {cycle_code: cycle.attributes.legacy_code})}">${cycle.attributes.name}</a></li>`);
+        } else if (incompleteCycleIds.has(cycle.id)) {
+          const incompleteListId = `${format.id}_cycles_${cycle.id}`;
+          if (!$(`#${incompleteListId}`).length) {
+            $(`#${format.id}_cycles`).append(`<li>${cycle.attributes.name}<ul id=${incompleteListId}></ul></li>`);
+          }
```
Comment (distributive) [inline app/Resources/views/Formats/formats.html.twig]: Currently this only affects Terminal Directive in eternal, which makes it stand out kinda weird. Having all cycles show their sets would be way too long though so I don't have a good solution of the top of my head.

**Review by distributive — COMMENTED** [review]
Would it be possible to add a few more comments between sections just to make it a little easier to see which section does what?

File: app/Resources/views/Formats/formats.html.twig (hunk 2 above)
Comment (plural) [reply app/Resources/views/Formats/formats.html.twig]: i think i can tidy the JS code up a bit for this, but i like the presentation - it's similar to what we do in the collection settings for decklist search and the deckbuilder.

File: app/Resources/views/Formats/formats.html.twig (hunk 1 above)
Comment (plural) [reply app/Resources/views/Formats/formats.html.twig]: sure.  easy enough since they are in one big box!

**Comment by plural** [comment]
I'll make a pass over the code to add some more comments and see if there is some more stuff to deduplicate.  It's annoying to have all the HTML be buried in the JS, but such is life.

The config stuff is in now as well, so I think this is generally good (and then will be useful for further PRs as well).

**Comment by plural** [comment]
allrighty, this should be good for a real review.

**Review by distributive — APPROVED** [review]

**Pull request merged**